services/refactorMessagesBuilder.js

```diff
@@ -41,15 +38,37 @@ class RefactorMessagesBuilder {
 
     _addMessage(role, text) {
         let msg = {};
-        if (this.useVertexAI) {
-            msg = {role: role, parts: [{text: text}]};
-        }
-        else {
-            msg = {role: role, content: text};
+
+        const modelTypes = {
+            "ChatGPT": (role, text) => ({role: role, content: text}),
+            "Vertex": (role, text) => ({role: role, parts: [{text: text}]}),
+            "ClaudeAi": (role, text) => ({role: role, content: JSON.stringify(text)}),
+            // Add more types as needed
         };
+        msg = modelTypes[config.openAiConfig.model_type](role, text);
         this.messages.push(msg);
     }
 
+    async _modelFactory(messages) {
```

Comment:
I suggest to put it in a new class called ModelAdapter and it will call the correct function based on this mapping + do the same for addMessage
Then the msgBuiilder will talk with the adapter and ill have one interface which he dont care about the types